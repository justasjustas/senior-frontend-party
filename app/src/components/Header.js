import React , { Component } from 'react';
import '../sass/header.scss';
import { withRouter } from 'react-router-dom';
import { unsetAuthorizationToken } from '../utils/authorization.js';

class Header extends Component {
  logOut = () => {
    unsetAuthorizationToken();
    this.props.history.push('/');
  }

  render() {
    return (
      <div className="row justify-content-between header-container">
        <div className="col-8">
          <img className="header-logo" src="static/images/logotype-testio-dark.png" alt="logo"/>
        </div>
        <div className="col-4">
          <div className="header-logout-container">
            <button onClick={this.logOut} className="btn btn-light header-logout-btn">
              <div className="header-logout-logo-container">
                <object type="image/svg+xml" data="static/images/ico-logout.svg">
                  icon
                </object>
              </div>
              Log Out
            </button>
          </div>
        </div>
      </div>
    )
  }
}

export default withRouter(Header);