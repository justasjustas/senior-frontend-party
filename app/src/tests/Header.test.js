import React from 'react';
import Header from '../components/Header';
import renderer from 'react-test-renderer';

it('Header renders correctly', () => {
  const tree = renderer
    .create(<Header.WrappedComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});