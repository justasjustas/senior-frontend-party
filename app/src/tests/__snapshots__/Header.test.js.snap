// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="row justify-content-between header-container"
>
  <div
    className="col-6 col-sm-8"
  >
    <img
      alt="logo"
      className="header-logo"
      src="static/images/logotype-testio-dark.png"
    />
  </div>
  <div
    className="col-6 col-sm-4"
  >
    <div
      className="header-logout-container"
    >
      <button
        className="btn btn-light header-logout-btn"
        onClick={[Function]}
      >
        <div
          className="header-logout-logo-container"
        >
          <object
            data="static/images/ico-logout.svg"
            type="image/svg+xml"
          >
            icon
          </object>
        </div>
        Log Out
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="row justify-content-between header-container"
>
  <div
    className="col-6 col-sm-8"
  >
    <img
      alt="logo"
      className="header-logo"
      src="static/images/logotype-testio-dark.png"
    />
  </div>
  <div
    className="col-6 col-sm-4"
  >
    <div
      className="header-logout-container"
    >
      <button
        className="btn btn-light header-logout-btn"
        onClick={[Function]}
      >
        <div
          className="header-logout-logo-container"
        >
          <object
            data="static/images/ico-logout.svg"
            type="image/svg+xml"
          >
            icon
          </object>
        </div>
        Log Out
      </button>
    </div>
  </div>
</div>
`;
